google/cloud/storage/bucket_test.cc

```diff
@@ -23,8 +23,12 @@ namespace {
 class MockClient : public storage::Client {
  public:
   using BucketGetResult = std::pair<storage::Status, storage::BucketMetadata>;
+  using ObjectInsertResult =
```

Comment:
Why not make a small struct or class containing these things?  It much more extensible if we find we ever need to add something.
I generally agree with this model, not just for C++: https://arne-mertz.de/2017/03/smelly-pair-tuple/

Reply:
I get the point about the smell, but this is no different than `std::set<T>::insert` returning a `std::pair<>`, it is a way to return multiple values from a function, we are not going to operate on the values all that much (maybe we could return the values by reference, but we lose RVO then).

Also, because with `std::pair<>` you get C++17 structured bindings, as in:

```C++
auto [k, v] = function();
```

With your own class you can get that, but you need to write the `get<int>(myclass&)` template, in the right namespace too.  We may not be using that, but our customers might.

Reply:
Fair enough.  Looks like most of the other client libraries just return the metadata outright, raising an exception if something goes wrong.  Since we can't always raise exceptions, this combo of status+metadata is fine.